src/Nest/Search/MultiSearch/MultiSearchTemplate/ElasticClient-MultiSearchTemplate.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Elasticsearch.Net;
+using Newtonsoft.Json;
+
+namespace Nest
+{
+	using System.Threading;
+	using MultiSearchTemplateCreator = Func<IApiCallDetails, Stream, MultiSearchResponse>;
+
+	public partial interface IElasticClient
+	{
+		/// <summary>
+		/// The multi search API allows to execute several search requests within the same API.
+		/// <para> </para><a href="http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-multi-search.html">http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-multi-search.html</a>
+		/// </summary>
+		/// <param name="selector">A descriptor that describes the search operations on the multi search api</param>
+		IMultiSearchResponse MultiSearchTemplate(Func<MultiSearchTemplateDescriptor, IMultiSearchTemplateRequest> selector);
+
+		/// <inheritdoc/>
+		IMultiSearchResponse MultiSearchTemplate(IMultiSearchTemplateRequest request);
+
+		/// <inheritdoc/>
+		Task<IMultiSearchResponse> MultiSearchTemplateAsync(Func<MultiSearchTemplateDescriptor, IMultiSearchTemplateRequest> selector, CancellationToken cancellationToken = default(CancellationToken));
+
+		/// <inheritdoc/>
+		Task<IMultiSearchResponse> MultiSearchTemplateAsync(IMultiSearchTemplateRequest request, CancellationToken cancellationToken = default(CancellationToken));
+	}
+
+	public partial class ElasticClient
+	{
+		/// <inheritdoc />
+		public IMultiSearchResponse MultiSearchTemplate(Func<MultiSearchTemplateDescriptor, IMultiSearchTemplateRequest> selector) =>
+			this.MultiSearchTemplate(selector?.Invoke(new MultiSearchTemplateDescriptor()));
+
+		/// <inheritdoc />
+		public IMultiSearchResponse MultiSearchTemplate(IMultiSearchTemplateRequest request)
+		{
+			return this.Dispatcher.Dispatch<IMultiSearchTemplateRequest, MultiSearchTemplateRequestParameters, MultiSearchResponse>(
+				request,
+				(p, d) =>
+				{
+					var converter = CreateMultiSearchTemplateDeserializer(p);
+					var serializer = this.ConnectionSettings.StatefulSerializer(converter);
+					var json = serializer.SerializeToBytes(p).Utf8String();
```

Comment:
`SerializeToString()`?

Does the custom serialization need to happen here, or can `p` be passed to  `this.LowLevelDispatch.MsearchTemplateDispatch<MultiSearchResponse>` for as the argument for the `PostData<object>` parameter (cast to `object`)?